android/src/main/java/com/cloudwebrtc/webrtc/GetUserMediaImpl.java

```diff
@@ -1062,4 +1063,16 @@ public class VideoCapturerInfo {
         int fps;
         boolean isScreenCapture = false;
     }
+
+    void setPreferredInputDevice(int i){
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+        } else {
+            android.media.AudioManager audioManager = ((android.media.AudioManager) applicationContext.getSystemService(Context.AUDIO_SERVICE));
+            final AudioDeviceInfo[] devices = audioManager.getDevices(android.media.AudioManager.GET_DEVICES_INPUTS);
+            if (devices.length>i){
+                audioDeviceModule.setPreferredInputDevice(devices[i]);
+            }
+        }
+    }
+
```

Comment:
Can you tell me which line it is, the purpose of this function is to try to set the priority audio source, so that you can choose the specified color audio device as audio input, such as bluetooth, built-in MIC, etc.

Reply:
Can you tell me which line it is, the purpose of this function is to try to set the priority audio source, so that you can choose to specify the audio device as audio input, such as bluetooth, built-in MIC, etc.

Reply:
This function needs to add an interface method at the Flutter level to call it.

Reply:
Yes, I understand the intent of this function. My current comment is completely about code formatting - the in lines 1077, 683 & 1158 you just add spare new lines.

Reply:
ok, i'll edit